infra/style/format.sh

```diff
@@ -6,13 +6,42 @@ set -e
 APPLY_PATCH_OPTION="-a"
 while [[ "$#" -gt 0 ]]; do
   case $1 in
-    --no-apply-patches) APPLY_PATCH_OPTION="";;
+    --no-apply-patches)
+      APPLY_PATCH_OPTION=""
+      shift
+      ;;
+    --diff-only)
+      CHECK_DIFF_ONLY="1"
+      shift
+      ;;
     *) echo "[ERROR] Unknown parameter passed: $1"; exit 255 ;;
   esac
-  shift
 done
 
 GIT_ROOT="$(git rev-parse --show-toplevel)"
 STYLE_DIR="${GIT_ROOT}/infra/style"
 
-lintrunner --force-color --all-files $APPLY_PATCH_OPTION --config "${GIT_ROOT}/.lintrunner.toml"
+if [[ "${CHECK_DIFF_ONLY}" = "1" ]]; then
+  MAIN_EXIST=$(git rev-parse --verify main)
+  CURRENT_BRANCH=$(git branch | grep \* | cut -d ' ' -f2-)
+  DIFF_COMMITS=`git log --graph --oneline main..HEAD | wc -l`
```

Comment:
When I run this with `./ccex format --diff-only`, I get this error.

```sh
fatal: ambiguous argument 'HEAD~22': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

When I run `git log --graph --oneline main..HEAD`, it generates very long logs (wc -l results in 22), when I only checked out this PR.

Reply:
Can you show your git log output?

Reply:
My main branch and the upstream/main branch had conflict. After I resolve the conflict, the problem's gone. ;-D